packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/CappedYieldDollarLongShortPairFinancialProductLibrary.sol

```diff
@@ -0,0 +1,85 @@
+ * will be worth < $1.
+ * Numerically this is found using:
+ * N = Notional of bond
+ * P = price of token
+ * T = number of tokens
+ * R1 = low price range
+ * C = collateral per pair, should be N/R1
+ * T = min(1,(R1/P)*C)
+ * If you want a yield dollar denominated as N = $1, you should set C to 1/R1. In that case, T = min(1,1/P).
+ * - At any price below the low price range (R1) the long side effectively holds a fixed number of collateral equal to
+ * collateralPerPair from the LSP with the value of expiryPercentLong = 1. This is the max payout in collateral.
+ * - Any price equal to or above R1 gives a payout equivalent to a yield dollar (bond) of notional N. In this range the
+ * expiryPercentLong shifts to keep the payout in dollar terms equal to the bond notional.
+ * With this equation, the contract deployer does not need to specify the bond notional N. The notional can be calculated
+ * by taking R1*collateralPerPair from the LSP.
+ */
+contract CappedYieldDollarLongShortPairFinancialProductLibrary is LongShortPairFinancialProductLibrary, Lockable {
+    using FixedPoint for FixedPoint.Unsigned;
+    using SignedSafeMath for int256;
+
+    mapping(address => uint256) public lowPriceRanges;
+
+    /**
+     * @notice Enables any address to set the low price range for an associated financial product.
+     * @param longShortPair address of the LSP contract.
+     * @param lowPriceRange low price range below which the payout transforms from a yield dollar to a short put option.
+     * @dev above the lowPriceRange the contract will payout a fixed amount of
+     * lowPriceRange*collateralPerPair (i.e the "notional" of the yield dollar).
+     * @dev Note: a) Any address can set these parameters b) existing LSP parameters for address not set.
+     * c) low price range can only be set once to prevent the deployer from changing the parameters after the fact.
+     * d) For safety, a low price range should be set before depositing any synthetic tokens in a liquidity pool.
+     * e) longShortPair must expose an expirationTimestamp method to validate it is correctly deployed.
+     */
+    function setLongShortPairParameters(address longShortPair, uint256 lowPriceRange) public nonReentrant() {
+        require(ExpiringContractInterface(longShortPair).expirationTimestamp() != 0, "Invalid LSP address");
+        require(lowPriceRanges[longShortPair] == 0, "Parameters already set");
+
+        lowPriceRanges[longShortPair] = lowPriceRange;
+    }
+
+    /**
+     * @notice Returns a number between 0 and 1e18 to indicate how much collateral each long and short token are
+     * entitled to per collateralPerPair.
+     * @param expiryPrice price from the optimistic oracle for the LSP price identifier.
+     * @return expiryPercentLong to indicate how much collateral should be sent between long and short tokens.
+     */
+    function percentageLongCollateralAtExpiry(int256 expiryPrice)
```

Comment:
why is `expiryPrice` allowed to be an `int` but `lowPriceRange` is a `uint`?

Reply:
Because it overrides this function in `LongShortPairFinancialProductLibrary`, which expects an `int`.

```
abstract contract LongShortPairFinancialProductLibrary {
    function percentageLongCollateralAtExpiry(int256 expiryPrice) public view virtual returns (uint256);
}
```